src/apps/advanced-search-v2/hooks/use-form-visibility.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from "react";
+import { useQueryState, parseAsJson, parseAsStringEnum } from "nuqs";
+import { buildCQLQuery, hasValidQuery } from "../lib/query-builder";
+import { isValidSuggestState, isValidFacetState } from "../lib/validation";
+
+type FormView = "search" | "results";
+
+interface UseFormVisibilityReturn {
+  view: FormView;
+  hasCurrentQuery: boolean;
+  showResults: boolean;
+  setView: (value: FormView) => Promise<URLSearchParams>;
+}
```

Comment:
I do not understand why we need to expose the `hasCurrentQuery` and `showResults` properties.

As I see it they are either duplicated by `FormView` or not used externally.